src/ch/fhnw/cpib/parser/Parser.java

```diff
@@ -116,60 +123,74 @@ private void decl() throws GrammarError {
             <recordShapeDecl>
          */
         if (token.terminal == Terminal.IDENT || token.terminal == Terminal.CHANGEMODE) {
-            stoDecl();
+            return stoDecl();
         } else if (token.terminal == Terminal.FUN) {
-            funDecl();
+            return funDecl();
         } else if (token.terminal == Terminal.PROC) {
-            procDecl();
+            return procDecl();
         } else if (token.terminal == Terminal.RECORD) {
-            recordShapeDecl();
+            return recordShapeDecl();
         } else throw new GrammarError("decl");
     }
 
-    private void recordShapeDecl() throws GrammarError {
+    private ConcSyn.ICpsTypedIdent cpsTypedIdent() throws GrammarError {
```

Comment:
Braucht es diese funktion überhaupt? kann man nicht einfach in ConcSyn.RecordShapeDecl zwei felder machen für typedIdent und repCommaTypedIdent?

Reply:
Wir haben ja diesen Schritt spezifisch aus der Grammatik entfernt weil er eigentlich unnötig ist. Ich fand es nur etwas hässlich, im record typedIdent und optTypedIdents zu haben wenn die ja eigentlich zusammen in einen anderen typen gehören. Das heisst dann logisch ändert sich nichts, nur im Baum sieht es dann etwas anders aus